core/controllers/tasks.py

```diff
@@ -76,6 +76,24 @@ def post(self):
 class SuggestionEmailHandler(base.BaseHandler):
     """Handler task of sending email of suggestion."""
 
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'POST': {
+            'exploration_id': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
+            },
+            'thread_id': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
```

Comment:
Why can this be `None`?

Reply:
Currently the test ```test_email_is_sent_when_suggestion_created``` fails when either ```exploration_id``` or ```thread_id``` is NOT defaulted to ```None```. It gives error that ```exploration_id```/```thread_id``` is missing within the handler args.

I Assume that it should default to ```None``` considering how other id-schemas are handled within this file. Such as ```contributor_user_id``` in the same file. But I may be wrong.

Reply:
Frontend tests where successful with the command ```python -m scripts.run_frontend_tests``` but I assume you mean manual verification through localhost? @vojtechjelinek

Reply:
I have verified the frontend for ```StateCompleteEventHandler´´´ and added it in the description of this PR (missed doing that before...). I Will add the proof for the other two tomorrow (hopefully) and notify you!

Reply:
No I meant that you need whether in the frontend code we actually call this handler with `None` or `null`

Reply:
After investigation this handler seems to not be called in the frontend? After asking about it on Gitter I got this response: 
![image](https://user-images.githubusercontent.com/60234650/194751152-44c02b03-4c82-4cd7-8ba3-5a12afbb18b6.png)
Not sure what to do going forward @vojtechjelinek

Reply:
@Nik-09 Hmm, I thought we still have the event handlers in the frontend, this PR seems to be working with just those and not improvements tab.

Reply:
Let's just remove the `SuggestionEmailHandler` and everything that relates to it, then.

Reply:
Done with removal @vojtechjelinek